src/k2/persistence/logStream/LogStream.cpp

```diff
@@ -0,0 +1,547 @@
+                while (payload.getDataRemaining() > 0){
+                    payload.read(logStreamName);
+                    payload.read(offset);
+                    payload.read(plogId);
+                    auto it = logStreamRecords.find(logStreamName);
+                    if (it == logStreamRecords.end()) {
+                        std::vector<dto::MetadataRecord> log;
+                        dto::MetadataRecord element{.plogId=plogId, .sealed_offset=0};
+                        log.push_back(std::move(element));
+                        logStreamRecords[logStreamName] = std::move(log);
+                    }
+                    else{
+                        it->second.back().sealed_offset = offset;
+                        dto::MetadataRecord element{.plogId=plogId, .sealed_offset=0};
+                        it->second.push_back(std::move(element));
+                    }
+                }
+
+                return seastar::do_with(std::move(logStreamRecords), [&] (auto& logStreamRecords){
+                    std::vector<seastar::future<std::tuple<Status, std::tuple<uint32_t, bool> > > > getStatusFutures;
+                    for (int logstreamName=LogStreamType::LogStreamTypeHead+1; logstreamName<LogStreamType::LogStreamTypeEnd; ++logstreamName){
+                        getStatusFutures.push_back(_logStreamMap[logstreamName]->get_plog_status(logStreamRecords[logstreamName].back().plogId));
+                    }
+                    return seastar::when_all_succeed(getStatusFutures.begin(), getStatusFutures.end())
+                    .then([&] (auto&& responses){
+                        Verb logstreamName=LogStreamType::LogStreamTypeHead;
+                        std::vector<seastar::future<Status> > reloadFutures;
+                        for (auto& response: responses){
+                            auto& [status, resp] = response;
+                            if (!status.is2xxOK()) {
+                                throw k2::dto::PlogGetStatusError("unable to obtain the information of a plog");
+                            }
+                            ++logstreamName;
+                            logStreamRecords[logstreamName].back().sealed_offset = std::get<0>(resp);
+
+                            reloadFutures.push_back(_logStreamMap[logstreamName]->reload(std::move(logStreamRecords[logstreamName])));
+                        }
+                        return seastar::when_all_succeed(reloadFutures.begin(), reloadFutures.end());
+                    })
+                    .then([this] (auto&& responses){
+                        for (auto& status: responses){
+                            if (!status.is2xxOK()) {
+                                throw k2::dto::LogStreamBaseReload("unable to reload the metadata");
+                            }
+                        }
+                        return seastar::make_ready_future<Status>(Statuses::S200_OK("successfully persist metadata"));
```

Comment:
I'm confused: why is a method "replay" persisting metadata?

Reply:
that is a typo. fixed the string here